src/app/test-record/visit/visit-edit/visit-edit.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
+import { TestStation } from '@app/models/test-station';
+import { ControlContainer, FormControl, FormGroupDirective, Validators } from '@angular/forms';
+import { TestResultModel } from '@app/models/test-result.model';
+import { TestRecordMapper } from '@app/test-record/test-record.mapper';
+
+@Component({
+  selector: 'vtm-visit-edit',
+  templateUrl: './visit-edit.component.html',
+  styleUrls: ['./visit-edit.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  viewProviders: [{ provide: ControlContainer, useExisting: FormGroupDirective }]
+})
+export class VisitEditComponent implements OnInit {
+  @Input() testRecord: TestResultModel;
+  @Input() testStations: TestStation[];
+  @Input() formErrors: string[];
+
+  testStationsOptions: string[];
+  testStationType = '';
+  testResultChildForm: FormGroupDirective;
+
+  constructor(private testRecordMapper: TestRecordMapper, parentForm: FormGroupDirective) {
+    this.testResultChildForm = parentForm;
+  }
+
+  ngOnInit() {
+    this.testResultChildForm.form.addControl(
+      'testStationNameNumber',
+      new FormControl(
+        this.testRecord.testStationName + ' (' + this.testRecord.testStationPNumber + ')',
+        Validators.required
+      )
+    );
+    this.testResultChildForm.form.addControl(
+      'testStationType',
+      new FormControl(
+        {
+          value: this.testRecord.testStationType
+            ? this.testRecord.testStationType.toUpperCase()
+            : '',
+          disabled: true
+        },
+        Validators.required
+      )
+    );
+
+    this.testResultChildForm.form.addControl(
+      'testerName',
+      new FormControl(this.testRecord.testerName, Validators.required)
+    );
+    this.testResultChildForm.form.addControl(
+      'testerEmailAddress',
+      new FormControl(this.testRecord.testerEmailAddress, [Validators.required, Validators.email])
+    );
+
+    this.testStationsOptions = !!this.testStations
+      ? this.testStations.map((res) => res.testStationName + ' (' + res.testStationPNumber + ')')
```

Comment:
ok it is the second time (or maybe more we have that, so this could be a util function if they are the same items in the array otherwise this could be simplified  as before

```